src/node/blockstorage.cpp

```diff
@@ -601,7 +601,9 @@ bool BlockManager::FindBlockPos(FlatFilePos& pos, unsigned int nAddSize, unsigne
 
     bool finalize_undo = false;
     if (!fKnown) {
-        while (m_blockfile_info[nFile].nSize + nAddSize >= (gArgs.GetBoolArg("-fastprune", false) ? 0x10000 /* 64kb */ : MAX_BLOCKFILE_SIZE)) {
+        const unsigned int max_blockfile_size{gArgs.GetBoolArg("-fastprune", false) ? 0x10000 /* 64kb */ : MAX_BLOCKFILE_SIZE};
+        assert(nAddSize < max_blockfile_size); // otherwise (possible in debug-only -fastprune mode) the while loop wouldn't terminate
```

Comment:
Instead of asserting, how about setting the block file size dynamically in case the block to be written exceeds the limit? Something like:
```
-        const unsigned int max_blockfile_size{gArgs.GetBoolArg("-fastprune", false) ? 0x10000 /* 64kb */ : MAX_BLOCKFILE_SIZE};
+        unsigned int max_blockfile_size = MAX_BLOCKFILE_SIZE;
+        if (gArgs.GetBoolArg("-fastprune", false)) {
+            max_blockfile_size = 0x10000; // 64kiB
+            if (nAddSize >= max_blockfile_size) {
+                max_blockfile_size = nAddSize + 1;
+            }
+        }
```

Reply:
Nice idea! I think that would work - I could do a full signet sync with it.

I'm still a bit undecided on general grounds, i.e. if it's okay to add more logic to accommodate a test-only mode that doesn't make sense for anything else but easier testing of pruning in the functional and unit tests. One could also argue that if no one but a developer would ever have a reason to use it, it'd be ok to just tell them more clearly not to create large blocks alongside using this option and abort otherwise?

@MarcoFalke : Do you have an opinion on this, or do you remember past discussions about test-only bitcoind options like this one?

Reply:
I switched to your suggestion, removing the assert!